external/dockerfile_functions.sh

```diff
@@ -451,7 +451,7 @@ print_clone_project() {
 
     echo -e "# Clone ${test} source" \
             "\nENV ${test_tag}=\$${test_tag}" \
-            "\nRUN git clone ${github_url}" \
+            "\nRUN git clone ${github_url} external_test_repo" \
```

Comment:
I believe this is a common script. Changing the folder name here means we change the behavior for all external tests - all external tests check out into `external_test_repo` folder. Therefore, all external tests need to cd into `external_test_repo` in their shell script, not just external_custom test. 

On the second thought, it would be easier to figure out the repo dir for external_custom. Can you get EXTERNAL_CUSTOM_REPO in test.sh? If so, just use cmd to figure out the folder (similar to 450).

Reply:
Since we have set the EXTERNAL_CUSTOM_TARGET as an environment variable, I think we can get the EXTERNAL_CUSTOM_REPO from the EXTERNAL_CUSTOM_TARGET using a function in test.sh itself.

Reply:
@lxia I have changed the logic of the external_test_repo as you told but one thing I noticed is that while moving into the jacoco folder, we have to cd into the following:[ https://github.com/adoptium/aqa-tests/blob/b766eb2c81252e8beefa3dcdf7e8c99c99210b68/external/jacoco/dockerfile/jacoco-test.sh#L41](https://github.com/adoptium/aqa-tests/blob/b766eb2c81252e8beefa3dcdf7e8c99c99210b68/external/jacoco/dockerfile/jacoco-test.sh#L41). So is it necessary to cd into the org.jacoco.build subfolder?

Reply:
interesting, so jacoco might be an exception case (not as suitable to use this external_custom approach as others, because it doesn't conform to the ideal scenario... where ideal is clone repository, cd into repo dir, run a single command to execute tests)

you can check if its necessary by seeing what happens if you cd into repo name (and not subfolder), are you still able to run tests or does it fail to work.

Reply:
@smlambert So I run the tests and it appears that we don't really need to cd into the org.jacoco.build folder again as the tests run successfully.
![sfd](https://user-images.githubusercontent.com/59281636/123460428-3d815a00-d605-11eb-9c78-231ff0d92aba.png)

Reply:
very good, thanks for confirming!

Reply:
Sorry for the misinformation i provided earlier! Although the make compile target can be run successfully, but the make _jacoco_test cannot be run without cd ing into the org.jacoco folder. So I think this cannot be included in the external_custom target tests.